mesh/mesh.cpp

```diff
@@ -6194,196 +6195,87 @@ void Mesh::LoadPatchTopo(std::istream &input, Array<int> &edge_to_knot)
-               v0 = ( v0 >= 0 ?  v0 : -v0-1);
-               v1 = ( v1 >= 0 ?  v1 : -v1-1);
-               if (v0 != v1)
-               {
-                  corrections++;
-                  if (v0 < v1)
-                  {
-                     edge_to_knot[e1] = (oedge[edge1[j]] >= 0 ? v0 : -v0-1);
-                  }
-                  else if (v1 < v0)
-                  {
-                     edge_to_knot[e0] = (oedge[edge0[j]] >= 0 ? v1 : -v1-1);
-                  }
-               }
-            }
-         }
-
-         passes++;
-      }
-      while (corrections > 0 && passes < GetNE() + 1);
-
-      // Check the validity of corrections applied
-      if (corrections > 0)
-      {
-         mfem::err<<"Edge_to_knot mapping potentially incorrect"<<endl;
-         mfem::err<<"  passes      = "<<passes<<endl;
-         mfem::err<<"  corrections = "<<corrections<<endl;
-      }
-
-      /* Renumber knotvectors, such that:
-         -- numbering is consecutive
-         -- starts at zero */
-      Array<int> cnt(NumOfEdges);
-      cnt = 0;
-      for (j = 0; j < NumOfEdges; j++)
-      {
-         k = edge_to_knot[j];
-         cnt[(k >= 0 ? k : -k-1)]++;
-      }
-
-      k = 0;
-      for (j = 0; j < cnt.Size(); j++)
-      {
-         cnt[j] = (cnt[j] > 0 ? k++ : -1);
-      }
-
-      for (j = 0; j < NumOfEdges; j++)
-      {
-         k = edge_to_knot[j];
-         edge_to_knot[j] = (k >= 0 ? cnt[k]:-cnt[-k-1]-1);
+         d = (i<8) ? ((i & 1) ? 1 : 0) : 2;
+         GetEdgeVertices(edges[i], v); // Get the vertices of edge i
+         // Not set yet -> use global index (p*dim+d)
+         kvidx = (kvs[d] == notset) ? p*dim+d : kvs[d];
+         // Sign is based on the edge's vertex indices
+         edge_to_pkv[edges[i]] = (v[1] > v[0]) ? kvidx : -1 - kvidx;
       }
+   }
 
-      // Print knot to edge mapping
-      mfem::out<<"Generated edge to knot mapping:"<<endl;
-      for (j = 0; j < NumOfEdges; j++)
-      {
-         int *v = edge_vertex->GetRow(j);
-         k = edge_to_knot[j];
-
-         v0 = v[0];
-         v1 = v[1];
-         if (k < 0)
-         {
-            v[0] = v1;
-            v[1] = v0;
-         }
-         mfem::out<<(k >= 0 ? k:-k-1)<<" "<< v[0] <<" "<<v[1]<<endl;
-      }
+   // ukv_to_pkv (unsigned) - unique knotvector index -> p*dim+d
+   ukv_to_pkv.SetSize(NumOfEdges);
+   for (int i = 0; i < NumOfEdges; i++)
+   {
+      ukv_to_pkv[i] = unsign(edge_to_pkv[i]);
+   }
+   ukv_to_pkv.Sort();
+   ukv_to_pkv.Unique();
 
-      // Terminate here upon failure after printing to have an idea of edge_to_knot.
-      if (corrections > 0 ) {mfem_error("Mesh::LoadPatchTopo");}
+   // Get edge_to_ukv = edge_to_pkv o ukv_to_pkv^{-1}
+   edge_to_ukv.SetSize(NumOfEdges);
+   bool sign;
+   int ukv, pkv;
+   for (int i = 0; i < NumOfEdges; i++)
+   {
+      sign = edge_to_pkv[i] < 0;
+      pkv = unsign(edge_to_pkv[i]);
+      ukv = ukv_to_pkv.Find(pkv);
+      MFEM_VERIFY(ukv >= 0, "Mapping error")
```

Comment:
This shouldn't be a bottleneck, so I guess it doesn't really matter, but instead of putting `MFEM_VERIFY` inside a loop, I usually set a `bool` flag and do just one `MFEM_VERIFY` on that flag, after the loop. This is assuming the verification has more than negligible cost, which probably isn't the case here.